File: app/src/main/java/kr/djspi/pipe01/tab/PlaneTab.java
package kr.djspi.pipe01.tab;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.google.gson.JsonObject;
import com.helloelliote.util.json.Json;

import kr.djspi.pipe01.R;

import static android.view.View.GONE;
import static android.view.View.VISIBLE;
import static kr.djspi.pipe01.BaseActivity.packageName;
import static kr.djspi.pipe01.BaseActivity.screenScale;

public class PlaneTab extends Fragment {

    private static final String TAG = PlaneTab.class.getSimpleName();
    private JsonObject jsonObject;
    private TextView tHorizontal, tVertical;
    private String resId;

    public PlaneTab() {
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof OnRecordListener) {
            OnRecordListener listener = (OnRecordListener) context;
            jsonObject = listener.getJsonObject();
        }
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.tab_plane, container, false);
        try {
            resId = Json.s(jsonObject, "file_plane").replace(".png", "_distance");
            ImageView imageView = view.findViewById(R.id.planeImageView);
            imageView.setImageResource(getResources().getIdentifier(resId, "drawable", packageName));
        } catch (UnsupportedOperationException | NullPointerException e) {
            view.findViewById(R.id.lay_0).setVisibility(GONE);
        }

        tHorizontal = view.findViewById(R.id.text_horizontal);
        tVertical = view.findViewById(R.id.text_vertical);
        tHorizontal.setText(Json.s(jsonObject, "horizontal"));
        tVertical.setText(Json.s(jsonObject, "vertical"));

        setPosition();

        return view;
    }

    private void setPosition() {
        // TODO: 2019-04-11 직상일때 0.00 칸 중간에 나오는 문제
        final int positionInt = Json.i(jsonObject, "position");
        if (Json.s(jsonObject, "shape").equals("직진형")) {
            switch (positionInt) {
                case 1:
                    setTranslation(true, false, 0.0f, -50.0f, 0.0f);
                    break;
                case 2:
                    if (resId.equals("plan_plate_str_2_out_distance")) {
                        setTranslation(true, true, 0.0f, 0.0f, 0.0f);
                    } else setTranslation(false, true, -50.0f, 0.0f, 0.0f);
                    break;
                case 3:
                    setTranslation(true, false, 0.0f, 50.0f, 0.0f);
                    break;
                case 4:
                    setTranslation(true, false, 0.0f, -100.0f, 0.0f);
                    break;
                case 5:
                    tHorizontal.setVisibility(GONE);
                    tVertical.setVisibility(GONE);
                    break;
                case 6:
                    setTranslation(true, false, 0.0f, 100.0f, 0.0f);
                    break;
                case 7:
                    setTranslation(true, false, 0.0f, -50.0f, 0.0f);
                    break;
                case 8:
                    if (resId.equals("plan_plate_str_8_out_distance")) {
                        setTranslation(true, true, 0.0f, 0.0f, 0.0f);
                    } else setTranslation(false, true, 50.0f, 0.0f, 0.0f);
                    break;
                case 9:
                    setTranslation(true, false, 0.0f, 50.0f, 0.0f);
                    break;
                default:
                    break;
            }
        } else {
            switch (positionInt) {
                case 1:
                    setTranslation(false, false, -100.0f, -170.0f, -350.0f);
                    break;
                case 2:
                    setTranslation(false, true, -100.0f, 0.0f, 0.0f);
                    break;
                case 3:
                    setTranslation(false, false, -100.0f, 175.0f, -350.0f);
                    break;
                case 4:
                    setTranslation(true, false, 0.0f, -90.0f, 0.0f);
                    break;
                case 5:
                    tHorizontal.setVisibility(GONE);
                    tVertical.setVisibility(GONE);
                    break;
                case 6:
                    setTranslation(true, false, 0.0f, 100.0f, 0.0f);
                    break;
                case 7:
                    setTranslation(false, false, 90.0f, -170.0f, 350.0f);
                    break;
                case 8:
                    setTranslation(false, true, 95.0f, 0.0f, 0.0f);
                    break;
                case 9:
                    setTranslation(false, false, 95.0f, 175.0f, 350.0f);
                    break;
                default:
                    break;
            }
        }
    }

    private void setTranslation(boolean noV, boolean noH, float vY, float hX, float hY) {
        if (noV) {
            tVertical.setVisibility(GONE);
            tHorizontal.setVisibility(VISIBLE);
        }
        if (noH) {
            tHorizontal.setVisibility(GONE);
            tVertical.setVisibility(VISIBLE);
        }
        tHorizontal.setTranslationX(hX * screenScale);
        tHorizontal.setTranslationY(hY * screenScale);
        tVertical.setTranslationY(vY * screenScale);
    }
}

File: app/src/main/java/kr/djspi/pipe01/tab/PlaneTab.kt
package kr.djspi.pipe01.tab

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.View.GONE
import android.view.View.VISIBLE
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.google.gson.JsonObject
import kr.djspi.pipe01.BaseActivity
import kr.djspi.pipe01.BaseActivity.Companion.screenRatio
import kr.djspi.pipe01.databinding.TabPlaneBinding
import kr.djspi.pipe01.geolocation.GeoTrans.CoodinateName.Companion.parseCoordinateName

class PlaneTab : Fragment() {

    private var _binding: TabPlaneBinding? = null
    private val binding get() = _binding!!
    private lateinit var json: JsonObject
    private lateinit var resId: String

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is OnRecordListener) {
            json = (context as OnRecordListener).jsonObject
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        _binding = TabPlaneBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        json["file_plane"].asString?.let {
            resId = it.replace(".png", "_distance")
            val imageView = binding.planeImageView
            imageView.setImageResource(
                resources.getIdentifier(
                    resId,
                    "drawable",
                    BaseActivity.defPackage
                )
            )
        }
        binding.textPlaneHorizontal.text = json["horizontal"].asString
        binding.textPlaneVertical.text = json["vertical"].asString
        setPosition()
        if (json["origin"] != null) {
            if (!json["origin"].isJsonNull) {
                binding.layTableSurveyPipe.apply {
                    visibility = VISIBLE
                    binding.textPlaneOrigin.text = parseCoordinateName(json["origin"].asString)
                    binding.textPlaneX.text = json["coordinate_x"].asString
                    binding.textPlaneY.text = json["coordinate_y"].asString
                }
            }
        }
        if (json["spi_origin"] != null) {
            if (!json["spi_origin"].isJsonNull) {
                binding.layTableSurveySpi.apply {
                    visibility = VISIBLE
                    binding.textPlaneOriginSpi.text = parseCoordinateName(json["spi_origin"].asString)
                    binding.textPlaneXSpi.text = json["spi_coordinate_x"].asString
                    binding.textPlaneYSpi.text = json["spi_coordinate_y"].asString
                }
            }
        }
//        val textView = view.findViewById<TextView>(R.id.lay_elb)
//        if (json["shape"].asString == PipeShape.PipeShapeEnum.엘보형135.type) {
//            textView.visibility = VISIBLE
//        } else {
//            textView.visibility = GONE
//        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    private fun setPosition() {
        val positionInt = json["position"].asInt
        when (json["shape"].asString) {
            "직진형" -> {
                when (positionInt) {
                    1 -> setTranslation(noV = true, vY = 0.0f, hX = -50.0f, hY = 0.0f)
                    2 -> {
                        if (resId == "plan_plate_str_2_out_distance") {
                            setTranslation(noV = true, noH = true, vY = 0.0f, hX = 0.0f, hY = 0.0f)
                        } else {
                            setTranslation(noH = true, vY = -50.0f, hX = 0.0f, hY = 0.0f)
                        }
                    }

                    3 -> setTranslation(noV = true, vY = 0.0f, hX = 50.0f, hY = 0.0f)
                    4 -> setTranslation(noV = true, vY = 0.0f, hX = -100.0f, hY = 0.0f)
                    5 -> {
                        binding.textPlaneHorizontal.visibility = GONE
                        binding.textPlaneVertical.visibility = GONE
                    }

                    6 -> setTranslation(noV = true, vY = 0.0f, hX = 100.0f, hY = 0.0f)
                    7 -> setTranslation(noV = true, vY = 0.0f, hX = -50.0f, hY = 0.0f)
                    8 -> {
                        if (resId == "plan_plate_str_8_out_distance") {
                            setTranslation(noV = true, noH = true, vY = 0.0f, hX = 0.0f, hY = 0.0f)
                        } else {
                            setTranslation(noH = true, vY = 50.0f, hX = 0.0f, hY = 0.0f)
                        }
                    }

                    9 -> setTranslation(noV = true, vY = 0.0f, hX = 50.0f, hY = 0.0f)
                }
            }

            "십자형" -> {
                when (positionInt) {
                    1 -> setTranslation(vY = -100.0f, hX = -170.0f, hY = -350.0f)
                    2 -> setTranslation(noH = true, vY = -50.0f, hX = 0.0f, hY = 0.0f)
                    3 -> setTranslation(vY = -100.0f, hX = 175.0f, hY = -350.0f)
                    4 -> setTranslation(noV = true, vY = 0.0f, hX = -90.0f, hY = 0.0f)
                    5 -> {
                        binding.textPlaneHorizontal.visibility = GONE
                        binding.textPlaneVertical.visibility = GONE
                    }

                    6 -> setTranslation(noV = true, vY = 0.0f, hX = 100.0f, hY = 0.0f)
                    7 -> setTranslation(vY = 90.0f, hX = -170.0f, hY = 350.0f)
                    8 -> setTranslation(noH = true, vY = 50.0f, hX = 0.0f, hY = 0.0f)
                    9 -> setTranslation(vY = 95.0f, hX = 175.0f, hY = 350.0f)
                }
            }

            else -> {
                when (positionInt) {
                    1 -> setTranslation(vY = -100.0f, hX = -170.0f, hY = -350.0f)
                    2 -> setTranslation(noH = true, vY = -100.0f, hX = 0.0f, hY = 0.0f)
                    3 -> setTranslation(vY = -100.0f, hX = 175.0f, hY = -350.0f)
                    4 -> setTranslation(noV = true, vY = 0.0f, hX = -90.0f, hY = 0.0f)
                    5 -> {
                        binding.textPlaneHorizontal.visibility = GONE
                        binding.textPlaneVertical.visibility = GONE
                    }

                    6 -> setTranslation(noV = true, vY = 0.0f, hX = 100.0f, hY = 0.0f)
                    7 -> setTranslation(vY = 90.0f, hX = -170.0f, hY = 350.0f)
                    8 -> setTranslation(noH = true, vY = 95.0f, hX = 0.0f, hY = 0.0f)
                    9 -> setTranslation(vY = 95.0f, hX = 175.0f, hY = 350.0f)
                }
            }
        }
    }

    private fun setTranslation(
        noV: Boolean = false,
        noH: Boolean = false,
        vY: Float,
        hX: Float,
        hY: Float
    ) {
        if (noV) {
            binding.textPlaneVertical.visibility = GONE
            binding.textPlaneHorizontal.visibility = VISIBLE
        }
        if (noH) {
            binding.textPlaneHorizontal.visibility = GONE
            binding.textPlaneVertical.visibility = VISIBLE
        }
        if (noV && noH) {
            binding.textPlaneHorizontal.visibility = GONE
            binding.textPlaneVertical.visibility = GONE
        }
        binding.textPlaneHorizontal.translationX = hX * screenRatio
        binding.textPlaneHorizontal.translationY = hY * screenRatio
        binding.textPlaneVertical.translationY = vY * screenRatio
    }
}
